src/pages/background/index.js

```diff
@@ -59,43 +70,32 @@ const handleContentScriptMessage = async (message, sender) => {
         return;
     }
     const url = await getCachedTabUrl(tabId); // 必然回复可用值
-    // console.log(message);
     if (message.type === "playlist") {
         const playlist = new Playlist({
             content: message.content,
             url: message.url,
             title: message.title,
             streamName: message.streamName
         });
+        if (url.includes("abema.tv")) {
+            for (const chunklist of playlist.chunkLists) {
+                chunklist.keyUrl = "abematv-license:";
+            }
+        }
         await Storage.addHistory(url, playlist, (p) => p.url === playlist.url);
     }
-    if (message.type === "chunklist") {
-        // 判断是否需要截获 key 而不是直接交由本体下载
-        if (message.keyUrl && needKeySites.some((site) => url.includes(site))) {
-            const keyIndexMap = {};
-            for (const playlist of await Storage.getHistory(url)) {
-                let modded = false;
-                for (const chunklist of playlist.chunkLists) {
-                    if (chunklist.keyIndex !== undefined) {
-                        keyIndexMap[chunklist.keyUrl] = chunklist.keyIndex;
-                        continue;
-                    }
-                    if (chunklist.url === message.url) {
-                        modded = true;
-                        chunklist.keyUrl = message.keyUrl;
-                        if (chunklist.keyUrl in keyIndexMap) {
-                            chunklist.keyIndex = keyIndexMap[chunklist.keyUrl];
-                        }
-                    }
-                }
-                if (!modded) {
-                    continue;
-                }
+    updateChunklist: if (message.type === "chunklist") {
```

Comment:
do not use label syntax

Reply:
It was a viable choice, as I wasn't misusing it like arbitrary goto. Definitely more concize than using an ad-hoc flag. Mozilla actually changed their docs to reflect their reevaluation of this, cf. https://stackoverflow.com/a/28003090.

Changed to using `while` and `Array.pop()`. Might be a little less efficient compared to the iterator approach (which is itself less efficient than indexing). I could have used a bare iterator (`playlists[Symbol.iterator]`) with its `next()` and `done`, but that syntax is even less well-known.

Reply:
Damn, `pop()` won't work. Changed back to a `for` loop.

Reply:
> It was a viable choice, as I wasn't misusing it like arbitrary goto. Definitely more concize than using an ad-hoc flag. Mozilla actually changed their docs to reflect their reevaluation of this, cf. https://stackoverflow.com/a/28003090.
> 
> Changed to using `while` and `Array.pop()`. Might be a little less efficient compared to the iterator approach (which is itself less efficient than indexing). I could have used a bare iterator (`playlists[Symbol.iterator]`) with its `next()` and `done`, but that syntax is even less well-known.

Efficiency is not the most important aspect in an open-source project unless it causes a performance issue. Writing clear and standard code is more beneficial for long-term maintenance. So I will not recommend any unusual syntax.